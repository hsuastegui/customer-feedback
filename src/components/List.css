.list {
    grid-column-start: 1;
    grid-column-end: span 2;
}

.listItem {
    margin-bottom: 5px;
    padding-bottom: 2px;
    border-bottom: 1px solid #232126;
    display: flex;
    align-items: center;

}

.listItem span:first-child {
    flex: 1;
}