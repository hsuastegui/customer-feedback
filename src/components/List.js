import React from "react";
import PropTypes from "prop-types";
import Rating from "./Rating";
import "./List.css";

function List({ data }) {
  return (
    <div className="list">
      {data.map((item) => (
        <div key={item.rating + item.name} className="listItem">
          <span>{item.comment}</span>
          <Rating rating={item.rating} />
        </div>
      ))}
    </div>
  );
}

List.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      email: PropTypes.string,
      rating: PropTypes.number,
      comment: PropTypes.string,
    }).isRequired
  ),
};

export default List;
