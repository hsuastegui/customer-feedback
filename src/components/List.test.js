import React from "react";
import { shallow } from "enzyme";
import List from "./List";

describe("List", () => {
  it("renders", () => {
    const data = [
      {
        name: "John",
        email: "john@email.com",
        rating: 3,
        comment: "Lorem ipsum",
      },
    ];
    const wrapper = shallow(<List data={data} />);
    expect(wrapper).toMatchSnapshot();
  });
});
