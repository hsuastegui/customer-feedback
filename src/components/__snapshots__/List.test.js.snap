// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders 1`] = `
<div
  className="list"
>
  <div
    className="listItem"
    key="3John"
  >
    <span>
      Lorem ipsum
    </span>
    <Stars
      rating={3}
      setRating={[Function]}
    />
  </div>
</div>
`;
